File: T9 Spelling/trash/t9.cpp
#include <iostream>
#include <stdio.h>
#include <string.h>

#define F0(i,n) for(i=0;i<n;i++)
#define F1(i,n) for(i=1;i<=n;i++)

using namespace std;

int main(int argc, char const *argv[])
{
	int i,j,k,l,m,n;
	string temp;
	cin>>n;
	getline(cin,temp);
	string kbd[10] = {" ","","abc","def","ghi","jkl","mno","pqrs","tuv","wxyz"};
	F1(i,n) {
		string str;
		string op = "";
		//cin>>str;
		getline(str);
		string prev_find;
		F0(j,str.length()) {
			F0(k,10) {
				int found = kbd[k].find(str[j]);
				if(found >= 0) {
					F0(l,found+1) op += (char)('0' + k);
					break;
				}
			}
		}
		cout<<"Case #"<<i<<": "<<op<<endl;
	}

	return 0;
}

File: T9 Spelling/T9.java
import java.util.*;

class T9 {
	public static String T9Code(String c) {
			String r = "";
			for(int i=0; i<c.length(); i++) {
				for(int j=0; j<T9.length; j++) {
					if(T9[j].contains(c.charAt(i)+"")) {
						if(r.endsWith(j+"")) r+=' ';
						for(int k=0;k<=T9[j].indexOf(c.charAt(i)); k++)
							r += j;
						break;
					}
				}
			}
			return r;
		}
	public static	String[] T9 = {" ","","abc","def","ghi","jkl","mno","pqrs","tuv","wxyz"};
		
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int n = in.nextInt();
		in.nextLine();
		
		for(int tt = 1; tt <= n; tt++) {
			String str = in.nextLine();
			out("Case #"+tt+": "+T9Code(str));
		}
	}
	public static void out(String str) {
		System.out.println(str);
	}
}